src/js/angular/core/templates/search-resource-input.html

```diff
@@ -1,13 +1,25 @@
-<div class="input-group input-group-lg">
+<div id="search-resource-box" class="input-group" ng-class="{'input-group-lg': !radioButtons}" style="white-space: nowrap">
     <input type="text" class="form-control view-res-input" placeholder="{{placeholder}}" ng-model="searchInput"
            ng-change="onChange()"
            ng-keydown="onKeyDown($event)">
-    <span class="input-group-btn">
-        <button ng-if="textButton !== ''" type="button" class="btn btn-primary autocomplete-text-btn" ng-click="checkIfValidAndSearchText()">{{textButton ? textButton : 'Text'}}</button>
+    <button ng-if="clearInputIcon && showClearInputIcon" class="btn btn-link clear-icon"
+          tooltip="Clear" tooltip-placement="bottom" tooltip-trigger="mouseenter">
+        <i class="icon-close" aria-hidden="true" ng-click="clearInput()"></i>
+    </button>
+    <span ng-if="!radioButtons" class="input-group-btn">
+        <button ng-if="textButton !== ''" type="button" class="btn btn-primary autocomplete-text-btn" ng-click="checkIfValidAndSearchText()">{{textButton ? textButton : 'Table'}}</button>
```

Comment:
`{{textButton ? textButton : 'Table'}}` these are better to be resolved in the controller